pkg/crc/preflight/preflight_daemon_task_check_windows.go

```diff
@@ -0,0 +1,138 @@
+</Task>
+`
+)
+
+func genDaemonTaskInstallTemplate(crcVersion, daemonServiceName string) string {
+	return fmt.Sprintf(daemonTaskTemplate,
+		crcVersion,
+		daemonServiceName,
+	)
+}
+
+func checkIfDaemonTaskInstalled() error {
+	_, stderr, err := powershell.Execute("Get-ScheduledTask", "-TaskName", constants.DaemonTaskName)
+	if err != nil {
+		logging.Debugf("%s task is not installed: %v : %s", constants.DaemonTaskName, err, stderr)
+		return err
+	}
+	if err := checkIfOlderTask(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func fixDaemonTaskInstalled() error {
+	// prepare the task script
+	binPath, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("unable to find the current executable location: %v", err)
+	}
+	binPathWithArgs := fmt.Sprintf("%s daemon", binPath)
+	var buf bytes.Buffer
+	if err := xml.EscapeText(&buf, []byte(binPathWithArgs)); err != nil {
+		return err
+	}
+	taskContent := genDaemonTaskInstallTemplate(
+		version.GetCRCVersion(),
+		buf.String(),
+	)
+
+	if _, stderr, err := powershell.Execute("Register-ScheduledTask", "-Xml", fmt.Sprintf(`'%s'`, taskContent), "-TaskName", constants.DaemonTaskName); err != nil {
+		return fmt.Errorf("failed to register %s task, %v: %s", constants.DaemonTaskName, err, stderr)
+	}
+
+	return nil
+}
+
+func removeDaemonTask() error {
+	if err := checkIfDaemonTaskRunning(); err == nil {
+		_, stderr, err := powershell.Execute("Stop-ScheduledTask", "-TaskName", constants.DaemonTaskName)
+		if err != nil {
+			logging.Debugf("unable to stop the %s task: %v : %s", constants.DaemonTaskName, err, stderr)
+			return err
+		}
+	}
+	if err := checkIfDaemonTaskInstalled(); err == nil {
+		_, stderr, err := powershell.Execute("Unregister-ScheduledTask", "-TaskName", constants.DaemonTaskName, "-Confirm:$false")
+		if err != nil {
+			logging.Debugf("unable to unregister the %s task: %v : %s", constants.DaemonTaskName, err, stderr)
+			return err
+		}
+	}
+	return nil
+}
+
+func checkIfDaemonTaskRunning() error {
+	stdout, stderr, err := powershell.Execute(fmt.Sprintf(`(Get-ScheduledTask -TaskName "%s").State`, constants.DaemonTaskName))
+	if err != nil {
+		logging.Debugf("%s task is not running: %v : %s", constants.DaemonTaskName, err, stderr)
+		return err
+	}
+	if strings.TrimSpace(stdout) != "Running" {
+		return fmt.Errorf("expected %s task to be in 'Running' but got '%s'", constants.DaemonTaskName, stdout)
+	}
+	return nil
+}
+
+func fixDaemonTaskRunning() error {
+	_, stderr, err := powershell.Execute("Start-ScheduledTask", "-TaskName", constants.DaemonTaskName)
+	if err != nil {
+		logging.Debugf("unable to run the %s task: %v : %s", constants.DaemonTaskName, err, stderr)
+		return err
+	}
+	return nil
+}
+
+func checkIfOlderTask() error {
+	stdout, stderr, err := powershell.Execute(fmt.Sprintf(`(Get-ScheduledTask -TaskName "%s").Version`, constants.DaemonTaskName))
+	if err != nil {
+		return fmt.Errorf("%s task is not running: %v : %s", constants.DaemonTaskName, err, stderr)
```

Comment:
Looking at the other `Get-ScheduledTask` calls the error message should be modified, its actually checking the `Version` here, and if this returns error then actually task is not installed ?

Reply:
@anjannath No, `Get-ScheduledTask` first check if the task is present and then this check if present task if running and if it is running then does the version is same.